File: go/1614.go
package leetcode

func maxDepth(s string) int {
	max := 0
	acc := 0

	for i := 0; i < len(s); i++ {
		switch s[i] {
		case '(':
			acc++
			if acc > max {
				max = acc
			}
		case ')':
			acc--
		}
	}

	return max
}

File: ts/1614.ts
function maxDepth(s: string): number {
  let ans = 0;
  let max = 0;

  for (let i = 0; i < s.length; i++) {
    if (s[i] === "(") ans++;
    else if (s[i] === ")") ans--;
    max = Math.max(max, ans);
  }

  return max;
}
